hazelcast/src/main/java/com/hazelcast/internal/config/YamlConfigSchemaValidator.java

```diff
@@ -95,6 +98,14 @@ public void validate(YamlMapping rootNode) {
             } else if (new HazelcastProperties(System.getProperties()).getBoolean(ROOT_LEVEL_INDENTATION_CHECK_ENABLED)) {
                 validateAdditionalProperties(rootNode, definedRootNodes.get(0));
             }
+            // Make the root node nullable by skipping validation when the value
+            // of root node is null. When changing the root element in the json schema
+            // to nullable, it significantly reduces the readability of the validation
+            // error messages, so we preferred this workaround.
+            if (rootNode != null && rootNode.child(definedRootNodes.get(0)) == null) {
+                return;
+            }
+
```

Comment:
I avoided changing the schema by adding this workaround since making an element nullable on the `json-schema` creates confusion in the messages of validation errors.

Reply:
When changing the `hazelcast` schema element to nullable, validator adds this cause into all kind of validation exceptions:
```
{
  "schemaLocation": "#/properties/hazelcast",
  "pointerToViolation": "#/hazelcast",
  "causingExceptions": [
    {
      "pointerToViolation": "#/hazelcast",
      "causingExceptions": [],
      "keyword": "type",
      "message": "expected: null, found: JSONObject"
    },
    ............
    ............
    }
  ],
  "keyword": "anyOf",
  "message": "#: no subschema matched out of the total 2 subschemas"
}
```